### ISSUE TITLE
[O# Parity] Support loading standalone projects

### ISSUE THREAD

**Issue by dibarbet** [dotnet/vscode-csharp#5722]
Currently the C# extension's project loading capabilities only supports loading full solution files.  O# supports loading csproj files so we likely need to as well.

**Comment by alexrp** [comment]
I see this is in "feature backlog", so I just want to ask what that means in terms of prioritization?

I use [Traversal](https://github.com/microsoft/MSBuildSdks/tree/main/src/Traversal) for basically every project I work on these days, which works great under OmniSharp since it can load loose project files, and in VS with Open Folder too. It would be unfortunate to have to take a step backwards and maintain solution files in order to use the new Roslyn-based LSP host. (FWIW, judging by the reactions on https://github.com/microsoft/vscode-dotnettools/issues/121, others are in the same boat.)

**Comment by v-Judy** [comment]
@dibarbet After installed the C# Dev Kit(`v 0.5.2`) with C# extension (`v 2.1.2`), when I want to run and debug the project, an error `Cannot create .NET debug configurations. The active C# project is not within folder` occurred.
Based on the [6050](https://github.com/dotnet/vscode-csharp/issues/6050#issuecomment-1668629016) comment, I try to add a solution and add the project to the solution, also set the dotnet.server.useOmnisharp option to get back the old behavior which can load projects, but it does not works for me.

If I only install the C# extension(`v 2.1.2`), this issue does not repro.

This issue only repro on Mac OS and does not repro on Linux and Windows OS. I'm not sure whether it's a same issue. Should I create a new issue to track it?

**Comment by jasonmalinowski** [comment]
@v-Judy Please file a new bug for this.

**Comment by v-Judy** [comment]
> @v-Judy Please file a new bug for this.

Thanks for your reply! I will file a new bug for it.

**Comment by vpetit-reimagine** [comment]
I have the same issue, without ever installing the `C# Dev Kit` extension but only the `C# v2.1.2`. I have a `.sln` at the root of the currently opened folder, and 2 projects in their own subfolders, referenced by the `.sln`.

This is utterly annoying that this is broken.

**Comment by jasonmalinowski** [comment]
> I have the same issue, without ever installing the `C# Dev Kit` extension but only the `C# v2.1.2`. I have a `.sln` at the root of the currently opened folder, and 2 projects in their own subfolders, referenced by the `.sln`.
> 
> This is utterly annoying that this is broken.

@vpetit-reimagine Can you file a new bug for us to track and with the details in our template? Sounds like this is different than the feature gap we're trying to track here.

**Comment by vpetit-reimagine** [comment]
> > I have the same issue, without ever installing the `C# Dev Kit` extension but only the `C# v2.1.2`. I have a `.sln` at the root of the currently opened folder, and 2 projects in their own subfolders, referenced by the `.sln`.
> > This is utterly annoying that this is broken.
> 
> @vpetit-reimagine Can you file a new bug for us to track and with the details in our template? Sounds like this is different than the feature gap we're trying to track here.

Here it is [#6403](https://github.com/dotnet/vscode-csharp/issues/6403).

**Comment by aelij** [comment]
I would love it if the C# extension would be able to open any MSBuild project file that contains project references (with or without the Traversal SDK - by utilizing the MSBuild static graph feature) and create a "dynamic solution" that is updated live when the project references change on disk (for example, a new .csproj file is created).

Solution files are cumbersome and hard to use, and we never create them manually. We currently have to rely on `slngen` which creates static solution files. When we pull from source control, we get restore errors due to missing projects in the .sln file on disk.

I've opened a [similar issue for VS](https://developercommunity.visualstudio.com/t/Open-MicrosoftBuildTraversal-project-f/1371244) which has gotten many votes.

**Comment by jasonmalinowski** [comment]
@aelij Yep, we're tracking that work with https://github.com/dotnet/vscode-csharp/issues/6008.
